### PR TITLE
feat(bedrockagentcore): support service/region in IAM credential provider for gateway targets

### LINKED ISSUE #37696: (bedrock-agentcore-alpha): support service/region in IAM credential provider for gateway targets
### Describe the feature

Add optional `service` / `region` arguments to `GatewayCredentialProvider.fromIamRole()`.

```ts
GatewayCredentialProvider.fromIamRole({ service: 'bedrock-runtime', region: 'us-east-1' });
```

### Use Case

coverage: missing feature

The L1 `IamCredentialProvider` (service / region) already exists, but the L2 alpha only emits `credentialProviderType: 'GATEWAY_IAM_ROLE'` and cannot specify the SigV4 signing service / region. Needed for cross-region calls and for targets where the service can't be inferred from the URL.

### Proposed Solution

Add optional props (`service?`, `region?`) to `fromIamRole()`. Zero-arg call keeps existing behavior.

### Other Information

_No response_

### Acknowledgements

- [x] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### AWS CDK Library version (aws-cdk-lib)

2.251.0

### AWS CDK CLI version

2.1119.0

### Environment details (OS name and version, etc.)

all

### PR DESCRIPTION
**Pull request by badmintoncryer**
### Issue # (if applicable)

Closes #37696.

### Reason for this change

`@aws-cdk/aws-bedrock-agentcore-alpha`'s `GatewayCredentialProvider.fromIamRole()` only emits `{ "CredentialProviderType": "GATEWAY_IAM_ROLE" }` and never populates the L1 `CredentialProvider.IamCredentialProvider` block (`Service`, `Region`). As a result, users cannot pin the SigV4 signing service / region for outbound calls.

The L1 already exposes `service` (required) and `region` (optional), so this is a coverage gap in the L2.

### Description of changes

- Add `GatewayIamRoleCredentialProviderProps` (`service?`, `region?`).
- `GatewayCredentialProvider.fromIamRole(props?)` now accepts the optional props and threads them into `GatewayIamRoleCredentialProviderConfig`.
- `_render()` now emits `credentialProvider.iamCredentialProvider: { service, region }` when props are set; the zero-arg call renders identically to today (backwards compatible).
- Synth-time validation (Token-aware) for length/pattern of `service` / `region`. If `region` is set without `service`, a `ValidationError` is raised since CFN requires `service` whenever the `IamCredentialProvider` block is present.
- Tighten `_render()` return type for this provider from `any` to `CfnGatewayTarget.CredentialProviderConfigurationProperty` to enforce the L1 property typing inside the alpha module. (Other providers in this module still return `any`; aligning them is out of scope for this PR.)

### Describe any new or updated permissions being added

None.

### Description of how you validated changes

add both unit and integ tests

### Checklist

- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by alvazjor** [comment]
Hi @badmintoncryer — heads up: `@aws-cdk/aws-bedrock-agentcore-alpha` has been graduated to stable in `aws-cdk-lib/aws-bedrockagentcore` (see #37876). This PR currently targets the alpha package which is now deprecated. Could you rebase against `main` and retarget your changes to `packages/aws-cdk-lib/aws-bedrockagentcore`? The feature is still relevant and we'd like to get it in. Let us know if you need help with the migration.

**Comment by badmintoncryer** [comment]
@alvazjor Thank you for your information. I've migrated this PR to stable module.

File: packages/aws-cdk-lib/aws-bedrockagentcore/lib/gateway/targets/target.ts [hunk 1]
```diff
@@ -644,6 +647,33 @@ export class GatewayTarget extends GatewayTargetBase implements IMcpGatewayTarge
...
+      return;
+    }
+    const hasExplicitIamCredentialProvider = this.credentialProviderConfigurations.some(
+      (c) => c instanceof GatewayIamRoleCredentialProviderConfig
+        && (c.service !== undefined || c.region !== undefined),
+    );
+    if (!hasExplicitIamCredentialProvider) {
+      return;
+    }
+    if (this.targetType !== McpTargetType.MCP_SERVER) {
```
Comment (alvazjor) [inline packages/aws-cdk-lib/aws-bedrockagentcore/lib/gateway/targets/target.ts]: This guard rejects OpenAPI targets, but AgentCore supports explicit service/region for OpenAPI too, not just MCP Server. The outbound-auth doc says the iamCredentialProvider block is used for "an MCP server or OpenAPI target," and only Lambda, API Gateway, and Smithy must omit it (https://docs.aws.amazon.com/bedrock-agentcore/latest/devguide/gateway-outbound-auth.html). As written, a valid OpenAPI + fromIamRole({ service, region }) config throws at synth.

Can you allow OpenAPI as well? Something like:

```ts
const iamCredentialProviderTargetTypes = [
  McpTargetType.MCP_SERVER,
  McpTargetType.OPENAPI_SCHEMA,
];
if (!iamCredentialProviderTargetTypes.includes(this.targetType)) {
  throw new ValidationError(
    lit`IamCredentialProviderUnsupportedTargetType`,
    `IamCredentialProvider with explicit service/region is only supported for MCP Server and OpenAPI targets, got target type "${this.targetType}"`,
    this,
  );
}
```

Please also add a unit test for an OpenAPI target accepting service/region.

File: packages/aws-cdk-lib/aws-bedrockagentcore/README.md [hunk 2]
```diff
@@ -1687,8 +1687,16 @@ and authorized during Inbound Auth.
...
+**IAM-based outbound authorization** – The gateway uses its execution role to authenticate with AWS services. This is the default and most common approach for Lambda targets and AWS service integrations. Use `GatewayCredentialProvider.fromIamRole()`; by default the gateway infers the SigV4 signing service and region from the target endpoint. For **MCP Server** targets, you can override either or both — useful for cross-region calls or when the service can't be inferred from the URL:
+
+```typescript fixture=default
+agentcore.GatewayCredentialProvider.fromIamRole({
+  service: 'bedrock-runtime', // SigV4 signing name (typically the endpoint prefix); see the AWS service authorization reference
+  region: 'us-east-1',         // defaults to the gateway's region
+});
+```
+
+The Bedrock AgentCore service only accepts `IamCredentialProvider` with explicit `service` / `region` for MCP Server targets. Lambda, Smithy, OpenAPI and API Gateway targets must use the bare `GatewayCredentialProvider.fromIamRole()` (with no arguments); the CDK enforces this with a synth-time validation.
```
Comment (alvazjor) [inline packages/aws-cdk-lib/aws-bedrockagentcore/README.md]: Same root cause as the target.ts guard: this line puts OpenAPI in the "must use the bare fromIamRole()" bucket, but OpenAPI targets do accept service/region per the outbound-auth doc. Please move OpenAPI into the allowed set so only Lambda, API Gateway, and Smithy require the no-arg call.

File: packages/aws-cdk-lib/aws-bedrockagentcore/test/agentcore/gateway/iam-credential-provider.test.ts [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+    test('skips validation for unresolved tokens', () => {
+      const tokenStack = new cdk.Stack();
+      expect(() => GatewayCredentialProvider.fromIamRole({
+        service: cdk.Token.asString({ resolve: () => 'bedrock-runtime' }),
+        region: tokenStack.region,
+      })).not.toThrow();
+    });
+
+    test('fails when explicit service/region is used with a Lambda target', () => {
```
Comment (alvazjor) [inline packages/aws-cdk-lib/aws-bedrockagentcore/test/agentcore/gateway/iam-credential-provider.test.ts]: The negative target-type test only covers Lambda. Once OpenAPI is allowed, could you add a positive test that an OpenAPI target renders IamCredentialProvider, plus negative coverage for API Gateway and Smithy via test.each? That pins the allowed/disallowed matrix so a future enum change can't regress it silently.

File: packages/aws-cdk-lib/aws-bedrockagentcore/lib/gateway/outbound-auth/iam-role.ts [hunk 4]
```diff
@@ -1,28 +1,91 @@
...
+   * Pattern: ^[a-zA-Z0-9-]+$
+   *
+   * @default - Gateway's own Region
+   */
+  readonly region?: string;
+}
+
 /**
  * Gateway IAM Role credential provider configuration implementation
  * Can be used with Lambda and Smithy targets
```
Comment (alvazjor) [inline packages/aws-cdk-lib/aws-bedrockagentcore/lib/gateway/outbound-auth/iam-role.ts]: This class comment still says "Can be used with Lambda and Smithy targets," which now reads wrong given explicit service/region is MCP-server/OpenAPI only (as defined here: https://docs.aws.amazon.com/bedrock-agentcore/latest/devguide/gateway-outbound-auth.html). Maybe: "Bare (no service/region) works with any target type; explicit service/region is only valid for MCP Server and OpenAPI targets."

File: packages/aws-cdk-lib/aws-bedrockagentcore/lib/gateway/targets/target.ts [hunk 5]
```diff
@@ -644,6 +647,33 @@ export class GatewayTarget extends GatewayTargetBase implements IMcpGatewayTarge
...
+   * (Lambda, Smithy, API Gateway and OpenAPI). Surface that as a synthesis-time error so users
+   * get fail-fast feedback.
+   */
+  private validateIamCredentialProviderTargetType(): void {
+    if (!this.credentialProviderConfigurations) {
+      return;
+    }
+    const hasExplicitIamCredentialProvider = this.credentialProviderConfigurations.some(
+      (c) => c instanceof GatewayIamRoleCredentialProviderConfig
+        && (c.service !== undefined || c.region !== undefined),
```
Comment (alvazjor) [inline packages/aws-cdk-lib/aws-bedrockagentcore/lib/gateway/targets/target.ts]: The config constructor already throws when region is set without service, so by the time this runs, region defined implies service defined. The `|| c.region !== undefined` branch is unreachable and `c.service !== undefined` is enough. Safe to leave, just slightly simpler without it.

**Review by alvazjor — DISMISSED** [review]
Thanks for the contribution, I added some comments to rework.

**Comment by badmintoncryer** [comment]
@alvazjor Thank you for your review!! I've addressed all of your comments.

File: packages/aws-cdk-lib/aws-bedrockagentcore/README.md [hunk 6]
```diff
@@ -1709,8 +1709,16 @@ and authorized during Inbound Auth.
 
 AgentCore Gateway supports the following types of outbound authorization:
 
-**IAM-based outbound authorization** – The gateway uses its execution role to authenticate with AWS services. This is the default
- and most common approach for Lambda targets and AWS service integrations.
+**IAM-based outbound authorization** – The gateway uses its execution role to authenticate with AWS services. This is the default and most common approach for Lambda targets and AWS service integrations. Use `GatewayCredentialProvider.fromIamRole()`; by default the gateway infers the SigV4 signing service and region from the target endpoint. For **MCP Server** and **OpenAPI** targets, you can override either or both — useful for cross-region calls or when the service can't be inferred from the URL:
```
Comment (alvazjor) [inline packages/aws-cdk-lib/aws-bedrockagentcore/README.md]: Nit: this says you can "override either or both" of `service` and `region`, but `region`-only isn't valid. The `IamCredentialProvider` block requires `service` (Required: Yes in the [Bedrock AgentCore Control API reference](https://docs.aws.amazon.com/bedrock-agentcore-control/latest/APIReference/API_IamCredentialProvider.html); `region` is optional and defaults to the gateway's Region), and the implementation already enforces this in `validate()` (`iam-role.ts`). Consider "you can override the service, and optionally the region too" so the doc matches the service contract. Non-blocking, but feels more clear that way

**Comment by badmintoncryer** [comment]
@alvazjor I've resolved it!

**Review by alvazjor — APPROVED** [review]
